### PR TITLE
fix: Add filter to pipelines/id/jobs for pr jobs only

### PR DESCRIPTION
**Pull request by tkyi**
## Context

Currently a lot of calls are being made on the PR tab page in the UI to: 
- `GET /pipelines/:id/events?count=1&page=1&prNum=prNum`
- `GET /jobs/:id/builds?count=10&page=1`
- `GET /events/:id/builds`

It would be nice if we could simplify these calls to get only latest builds in a PR and make fewer calls to the API.

## Objective

This PR adds a filter to an endpoint `GET /pipelines/:id/jobs?type=pr` to get pr jobs only for a pipeline.

## References
Related to https://github.com/screwdriver-cd/screwdriver/pull/3070, https://github.com/screwdriver-cd/datastore-sequelize/pull/93

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by jithine** [comment]
How can a client ask for all jobs except PR ?

File: plugins/pipelines/README.md [hunk 1]
```diff
@@ -107,9 +107,9 @@ Only PR events of specified PR number will be searched when `prNum` is set
 `GET /pipelines/{id}/events?page={pageNumber}&count={countNumber}&sort={sort}&prNum={prNumber}`
 
 #### Get all jobs (including pull requests jobs)
-`archived` is optional and has a default value of `false`, which makes the endpoint not return archived jobs (e.g. closed pull requests)
+`archived` is optional and has a default value of `false`, which makes the endpoint not return archived jobs (e.g. closed pull requests). `type` is optional and can be set to `pr` to only return PR jobs.
 
-`GET /pipelines/{id}/jobs?archived={boolean}`
+`GET /pipelines/{id}/jobs?archived={boolean}&type=pr`
```
Comment (VonnyJap) [inline plugins/pipelines/README.md:112]: What about having a param as per our original discussion?
`v4/pipelines/:id/jobs?pr=true`
`pr=true` returns all jobs in the pipeline that belong to a PR
`pr=false` returns all jobs in the pipeline that belong to the main branch of the SCM repo
when the `pr` parameter is not set, the response should be to return all jobs for the pipeline

**Comment by tkyi** [comment]
@jithine 
> How can a client ask for all jobs except PR ?

was thinking query param `type=pipeline`/`type=pr`

File: plugins/pipelines/README.md [hunk 2]
```diff
@@ -106,10 +106,15 @@ Only PR events of specified PR number will be searched when `prNum` is set
...
-#### Get all jobs (including pull requests jobs)
-`archived` is optional and has a default value of `false`, which makes the endpoint not return archived jobs (e.g. closed pull requests)
+#### Get all jobs
 
-`GET /pipelines/{id}/jobs?archived={boolean}`
+Arguments:
+
+* `archived` - Optional and has a default value of `false`, which makes the endpoint not return archived jobs (e.g. closed pull requests)
+* `type` - Optional and can be set to `pr` or `pipeline` to only return PR jobs or non-PR jobs
+* `jobName` - Optional and can be set to only return only a single job
```
Comment (jithine) [inline plugins/pipelines/README.md:115]: What's the use case for this param ? is this a single job ? or does it match multiple ? 

won't it be ideal for this to be a path param ?

Comment (tkyi) [reply plugins/pipelines/README.md:115]: https://github.com/screwdriver-cd/screwdriver/issues/1522

**Review by VonnyJap — APPROVED** [review]

**Pull request merged**